### PR TITLE
[zh-cn]: Update the translation of Array.map()

### PR DESCRIPTION
**Pull request by JasonLamv-t**
### Description

Update the translation of `Array.map()` method.

### Motivation

Ditto.

### Additional details

Nope.

### Related issues and pull requests

#12761

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/global_objects/array/map/index.md [hunk 1]
```diff
@@ -135,46 +157,48 @@ const values = Array.prototype.map.call(elems, ({ value }) => value);
...
 
-['1', '2', '3'].map(returnInt); // [1, 2, 3]
+["1", "2", "3"].map(returnInt); // [1, 2, 3]
 // 实际结果是一个数字数组（如预期）
 
 // 与上面相同，但使用简洁的箭头函数语法
-['1', '2', '3'].map((str) => parseInt(str)); // [1, 2, 3]
+["1", "2", "3"].map((str) => parseInt(str)); // [1, 2, 3]
 
 // A simpler way to achieve the above, while avoiding the "gotcha":
```
Comment (jasonren0403) [inline files/zh-cn/web/javascript/reference/global_objects/array/map/index.md:188]: 试着翻译了下这个
```suggestion
// 实现上述目标更简单的方法，同时避免了“骗招”：
```

File: files/zh-cn/web/javascript/reference/global_objects/array/map/index.md [hunk 2]
```diff
@@ -81,51 +78,76 @@ const kvArray = [
...
+  0: 2,
+  1: 3,
+  2: 4,
+};
+console.log(Array.prototype.map.call(arrayLike, (x) => x ** 2));
+// [ 4, 9, 16 ]
 ```
 
-### querySelectorAll 应用
+### 在 NodeList 上可以通用地使用 map()
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/array/map/index.md:117]:
```suggestion
### 在 NodeList 上通用地使用 map()
```

File: files/zh-cn/web/javascript/reference/global_objects/array/map/index.md [hunk 3]
```diff
@@ -135,46 +157,48 @@ const values = Array.prototype.map.call(elems, ({ value }) => value);
 
 我们期望输出 `[1, 2, 3]`, 而实际结果是 `[1, NaN, NaN]`.
 
-parseInt 经常被带着一个参数使用，但是这里接受两个。第一个参数是一个表达式而第二个是回调函数的基，`Array.prototype.map` 传递 3 个参数：
+`parseInt` 函数通常只使用一个参数，但其实可以传入两个参数。第一个参数是表达式，第二个参数是解析该表达式的基数。当在 `Array.prototype.map` 的回调函数中使用 `parseInt` 函数时，`map` 方法会传递3个参数：
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/array/map/index.md:160]:
```suggestion
{{jsxref("parseInt")}} 函数通常只使用一个参数，但其实可以传入两个参数。第一个参数是表达式，第二个参数是解析该表达式的基数。当在 `Array.prototype.map` 的回调函数中使用 `parseInt` 函数时，`map` 方法会传递3个参数：
```

File: files/zh-cn/web/javascript/reference/global_objects/array/map/index.md [hunk 4]
```diff
@@ -135,46 +157,48 @@ const values = Array.prototype.map.call(elems, ({ value }) => value);
...
 
 - the element
 - the index
 - the array
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/array/map/index.md:164]: 翻译一下

```suggestion
- 元素
- 索引
- 数组
```

File: files/zh-cn/web/javascript/reference/global_objects/array/map/index.md (hunk 3 above)
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/map/index.md:160]:
```suggestion
{{jsxref("parseInt")}} 函数通常只使用一个参数，但其实可以传入两个参数。第一个参数是表达式，第二个参数是解析该表达式的基数。当在 `Array.prototype.map` 的回调函数中使用 `parseInt` 函数时，`map` 方法会传递 3 个参数：
```

**Review by yin1999 — APPROVED** [review]

**Pull request merged**